funnel/models/rsvp.py

```diff
@@ -89,8 +105,27 @@ def rsvp_yes(self):
         message=__("Your response has been saved"),
         type='dark',
     )
-    def rsvp_no(self):
-        pass
+    def rsvp_no(self, actor):
+        previous_membership = (
+            CommentsetMembership.query.filter(CommentsetMembership.is_active)
+            .filter_by(
+                commentset=self.project.commentset,
+                user_id=actor.id,
+            )
+            .one_or_none()
+        )
+        if previous_membership is None:
+            # just in case there is no membership record when a user is canceling
+            # registration. Cna happen right after introducing the membership feature.
+            # In that case, create one and then revoke it to have a record.
+            previous_membership = CommentsetMembership(
+                commentset=self.project.commentset,
+                user=actor,
+                is_subscriber=False,
+            )
+            db.session.add(previous_membership)
+
+        previous_membership.revoke(actor=actor)
```

Comment:
This adds a membership and then revokes it immediately. Why? It should be enough to revoke existing subscription, if any.